### PR TITLE
ADAPT-3124: Search styles

### PR DESCRIPTION
**Pull request by Lyafel**
# Summary
Various styling fixes for search

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to /search
3. Verify changes to the search page and search overlay mentioned in the ticket

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/search/searchModal.js [hunk 1]
```diff
@@ -27,20 +27,22 @@ const SearchModal = ({ isOpen, setIsOpen, onClose }) => {
...
       <div className="su-max-w-1000 su-mx-auto">
-        <Heading
-          font="serif"
-          size={2}
-          level={2}
-          className="su-text-white su-text-center"
-        >
-          Hello, what can we help you find today?
-        </Heading>
+        <div className="su-mb-108">
```
Comment (yvonnetangsu) [inline src/components/search/searchModal.js:30]:
```suggestion
        <div className="su-rs-mb-4">
```
Please use responsive-spacing su-rs-______ classes for these larger spacings

File: src/components/search/searchModal.js [hunk 2]
```diff
@@ -27,20 +27,22 @@ const SearchModal = ({ isOpen, setIsOpen, onClose }) => {
...
+            Hello, what can we help you find today?
+          </Heading>
+        </div>
         <SearchFieldModal
           ref={searchFieldRef}
           onSubmit={() => setIsOpen(false)}
         />
         {story && content && (
-          <div className="su-mt-108">
+          <div className="su-mt-78">
```
Comment (yvonnetangsu) [inline src/components/search/searchModal.js:45]: Please use responsive-spacing here also

Comment (Lyafel) [reply src/components/search/searchModal.js]: Removed. Looks like it's configurable in storyblok but there's also a different ticket for this

**Comment by sherakama** [comment]
@yvonnetangsu can you resolve the merge conflict here. I am unsure which styles are correct in this PR.

**Review by yvonnetangsu — CHANGES_REQUESTED** [review]
@Lyafel For this ticket, looks like there are a couple of items still:

- Search red button needs to be circular (looks like it’s taller vertically than horizontally currently)

- Magnifying glass submit button should be `su-digital-red-xlight` (#F83535) on-hover/focus

we have some spacing/font size clean up to do, but that could be done in the other ticket.

File: src/components/search/searchFieldModal.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ const SearchFieldModal = React.forwardRef((props, ref) => {
   const inputClasses = `su-border-0 su-bg-transparent su-text-black-10 su-text-black-40::placeholder su-w-full su-flex-1
   su-rs-px-1 su-py-10 su-text-m4 focus:su-outline-none focus:su-ring-0 focus:su-ring-transparent`;
 
-  const submitBtnClasses = `su-w-70 su-h-70 su-rounded-full su-transition-colors su-bg-digital-red hocus:su-bg-black
+  const submitBtnClasses = `su-min-w-[70px] su-w-70 su-h-70 su-rounded-full su-transition-colors su-bg-digital-red-xlight hocus:su-bg-black
```
Comment (yvonnetangsu) [inline src/components/search/searchFieldModal.js:23]: We'd like the normal state of the button to be digital-red, and only digital-red-xlight on hover/focus.

**Review by yvonnetangsu — CHANGES_REQUESTED** [review]
Please revise button color.

**Comment by Lyafel** [comment]
@yvonnetangsu Sorry I missed that.. Changed

**Review by yvonnetangsu — APPROVED** [review]
GTG

File: src/components/page-types/searchPage.js [hunk 4]
```diff
@@ -211,10 +213,21 @@ const SearchPage = (props) => {
...
+
+  useEffect(() => {
+    const params = qs.parse(search);
+    setPageParam(undefined);
+    setQueryParam(params.q || undefined);
+    setPage(0);
+    setQuery(params.q);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search]);
+
```
Comment (yvonnetangsu) [inline src/components/page-types/searchPage.js:226]: @sherakama Does this look good to you?

Comment (yvonnetangsu) [reply src/components/page-types/searchPage.js:226]: @sherakama I'm going to merge this in so I can continue to work on the UI tweaks for search. I tested this and does seem to me that it's working as expected now. Please review retro when you have time 😄

**Pull request merged**